### PR TITLE
Fix compilation with gcc 4.9.4

### LINKED ISSUE #6030: Compilation error with gcc 4.9.4: "error: unknown option after ‘#pragma GCC diagnostic’ kind"
**OS:** Gentoo Linux x86_64
**Version:** gcc 4.9.4
**Commit/Build:** v0.1.0

Compiling, I get the following errors:
```
In file included from /var/tmp/portage/games-simulation/openrct2-0.1.0/work/OpenRCT2-0.1.0/src/openrct2/drawing/X8DrawingEngine.cpp:27:0:
/var/tmp/portage/games-simulation/openrct2-0.1.0/work/OpenRCT2-0.1.0/src/openrct2/drawing/X8DrawingEngine.h:63:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
 #pragma GCC diagnostic ignored "-Wsuggest-final-types"
                                ^
/var/tmp/portage/games-simulation/openrct2-0.1.0/work/OpenRCT2-0.1.0/src/openrct2/drawing/X8DrawingEngine.cpp:137:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
 #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
                                ^
cc1plus: all warnings being treated as errors
```

These options are not supported by gcc 4.9.4 yet...

### PR DESCRIPTION
**Pull request by olifre**
This fixes  #6030  which enables compilation with gcc 4.9.4 again. 
It's also more correct: The special warning types are now only disabled via pragmas if the compiler was tested to actually support them.

### CODE REVIEW

File: src/openrct2/drawing/X8DrawingEngine.cpp [hunk 1]
```diff
@@ -133,8 +133,10 @@ void X8RainDrawer::Restore()
     }
 }
 
+#ifdef __WARN_SUGGEST_FINAL_TYPES__
```
Comment (IntelOrca) [inline src/openrct2/drawing/X8DrawingEngine.cpp]: `__WARN_SUGGEST_FINAL_METHODS__`?

Comment (olifre) [inline src/openrct2/drawing/X8DrawingEngine.cpp]: Wuah! Yes, indeed, sorry, thanks for reviewing carefully. Will re-push in a minute.

File: src/openrct2/core/Guard.hpp [hunk 2]
```diff
@@ -23,6 +23,7 @@ extern "C" {
 #endif // __cplusplus
 
 void openrct2_assert_va(bool expression, const char * message, va_list va);
+inline void openrct2_assert(bool expression, const char * message, ...);
```
Comment (Gymnasiast) [inline src/openrct2/core/Guard.hpp]: Why did you add this line?

Comment (olifre) [inline src/openrct2/core/Guard.hpp]: Without that line, you don't comply to C standard, I believe. GCC 4.9.4 says:
```
In file included from /home/olifre/gits/OpenRCT2/src/openrct2/drawing/drawing.c:19:0:
/home/olifre/gits/OpenRCT2/src/openrct2/drawing/../core/Guard.hpp:26:13: error: no previous declaration for ‘openrct2_assert’ [-Werror=missing-declarations]
 inline void openrct2_assert(bool expression, const char * message, ...)
             ^
cc1: all warnings being treated as errors
```
I guess the support for having only the definition and to completely omit the declaration is some gnu11 extension, but I don't think this is in any standard - or is it?

Comment (IntelOrca) [inline src/openrct2/core/Guard.hpp]: Perhaps it needs to be `static inline`?

Comment (olifre) [inline src/openrct2/core/Guard.hpp]: `static inline` would of course also work, then also the explicit declaration is not required. Would you prefer this over the "normal" `inline` (which definitely needs the declaration)? If so, I can certainly adapt the PR accordingly.

Comment (janisozaur) [inline src/openrct2/core/Guard.hpp]: use `static`

Comment (olifre) [inline src/openrct2/core/Guard.hpp]: @janisozaur Ok, done and re-pushed the updated commits.

**Pull request merged**